src/Control/Effect/Carrier.hs

```diff
@@ -52,3 +63,31 @@ handlePure handler = hmap handler . fmap' handler
 handleCoercible :: (HFunctor sig, Coercible f g) => sig f (f a) -> sig g (g a)
 handleCoercible = handlePure coerce
 {-# INLINE handleCoercible #-}
+
+{-# DEPRECATED handleReader, handleState, handleEither, handleTraversable
+  "Compose carrier types from other carriers and define 'eff' with handleCoercible instead" #-}
```

Comment:
May also note that `handle`/`handlePure` can be used for non-compositional carriers.

Reply:
@patrickt: I think this got missed; what do you think about mentioning `handle`/`handlePure` here?